src/WebJobs.Script/OutOfProc/Http/DefaultHttpWorkerService.cs

```diff
@@ -0,0 +1,178 @@
+        {
+            _httpClient = httpClient;
+            _httpWorkerOptions = httpWorkerOptions.Value;
+            _logger = loggerFactory.CreateLogger<DefaultHttpWorkerService>();
+        }
+
+        public Task InvokeAsync(ScriptInvocationContext scriptInvocationContext)
+        {
+            if (scriptInvocationContext.FunctionMetadata.IsSimpleHttpTriggerFunction())
+            {
+                return ProcessHttpInAndOutInvocationRequest(scriptInvocationContext);
+            }
+            return ProcessDefaultInvocationRequest(scriptInvocationContext);
+        }
+
+        internal async Task ProcessHttpInAndOutInvocationRequest(ScriptInvocationContext scriptInvocationContext)
+        {
+            _logger.LogDebug("Will invoke simple httpTrigger function:{functionName} invocationId:{invocationId}", scriptInvocationContext.FunctionMetadata.Name, scriptInvocationContext.ExecutionContext.InvocationId);
+
+            ScriptInvocationResult scriptInvocationResult = new ScriptInvocationResult()
+            {
+                Outputs = new Dictionary<string, object>()
+            };
+
+            HttpRequestMessage httpRequestMessage = null;
+
+            (string name, DataType type, object val) input = scriptInvocationContext.Inputs.First();
+
+            HttpRequest httpRequest = (HttpRequest)input.val;
+            if (httpRequest == null)
+            {
+                throw new ArgumentNullException(nameof(httpRequest));
+            }
+
+            try
+            {
+                // Build HttpRequestMessage from HttpTrigger binding
+                HttpRequestMessageFeature httpRequestMessageFeature = new HttpRequestMessageFeature(httpRequest.HttpContext);
+                httpRequestMessage = httpRequestMessageFeature.HttpRequestMessage;
+
+                AddRequestHeadersAndSetRequestUri(httpRequestMessage, scriptInvocationContext.FunctionMetadata.Name, scriptInvocationContext.ExecutionContext.InvocationId.ToString());
+
+                // Populate query params from httpTrigger
+                string httpWorkerUri = QueryHelpers.AddQueryString(httpRequestMessage.RequestUri.ToString(), httpRequest.GetQueryCollectionAsDictionary());
+                httpRequestMessage.RequestUri = new Uri(httpWorkerUri);
+
+                _logger.LogDebug("Sending http request message for simple httpTrigger function:{functionName} invocationId:{invocationId}", scriptInvocationContext.FunctionMetadata.Name, scriptInvocationContext.ExecutionContext.InvocationId);
+
+                HttpResponseMessage invocationResponse = await _httpClient.SendAsync(httpRequestMessage);
+
+                _logger.LogDebug("Received http response for simple httpTrigger function:{functionName} invocationId:{invocationId}", scriptInvocationContext.FunctionMetadata.Name, scriptInvocationContext.ExecutionContext.InvocationId);
```

Comment:
All of these logger messages should move to `LoggerMessage.Define`s. And same comments about formatting as above.